browser/ui/ai_chat/ai_chat_agent_profile_helper.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/ai_chat/ai_chat_agent_profile_helper.h"
+
+#include "base/path_service.h"
+#include "brave/components/ai_chat/core/browser/utils.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#include "brave/components/constants/brave_constants.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile_attributes_storage.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/browser/profiles/profile_window.h"
+#include "chrome/browser/themes/theme_service.h"
+#include "chrome/browser/themes/theme_service_factory.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_window/public/browser_window_features.h"
+#include "chrome/browser/ui/views/side_panel/side_panel_ui.h"
+#include "chrome/common/chrome_paths.h"
+
+namespace ai_chat {
+
+namespace {
+
+const SkColor kAIChatAgentProfileThemeColor = SkColorSetRGB(253, 58, 122);
+const char16_t kAIChatAgentProfileName[] = u"Leo AI Content Agent";
+
+void SetupAndOpenAIChatAgentProfile(base::OnceCallback<void(Browser*)> callback,
+                                    Profile* profile) {
+  // This runs for every profile open so that we can update
+  // with any changes to the profile.
+
+  // Set theme
+  auto* theme_service = ThemeServiceFactory::GetForProfile(profile);
+  theme_service->SetUserColor(kAIChatAgentProfileThemeColor);
+  // Assume user has opted-in in some profile already in order to get
+  // here, so we can copy that preference.
+  ai_chat::SetUserOptedIn(profile->GetPrefs(), true);
+  ProfileManager* profile_manager = g_browser_process->profile_manager();
+  ProfileAttributesStorage& storage =
+      profile_manager->GetProfileAttributesStorage();
+  ProfileAttributesEntry* attributes =
+      storage.GetProfileAttributesWithPath(profile->GetPath());
+  attributes->SetLocalProfileName(kAIChatAgentProfileName, false);
+
+  // Open browser window
+  profiles::OpenBrowserWindowForProfile(
+      base::BindOnce(
+          [](base::OnceCallback<void(Browser*)> callback, Browser* browser) {
+            // Open sidebar
+            browser->GetFeatures().side_panel_ui()->Show(
+                SidePanelEntryId::kChatUI);
+            std::move(callback).Run(browser);
+          },
+          std::move(callback)),
+      false, false, false, profile);
+}
+
+}  // namespace
+
+void OpenBrowserWindowForAIChatAgentProfile(
+    Profile* from_profile,
```

Comment:
It looks this function and the one above need a non-null pointer to profile, so it would be better to just use `Profile&` here, and eliminate the check below.

If the caller to these functions have only a pointer, that you don't know to be valid or not, use `CHECK_DEREF` in the caller if you're not sure if the dereferencing could be iffy.